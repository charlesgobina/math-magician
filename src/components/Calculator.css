.calc-btns-wrapper {
  background-color: #bdbdbd;
  margin-left: 2px;
}

.display {
  width: 100%;
}

.display input {
  width: 100%;
  height: 60px;
  font-size: 30px;
  text-align: right;
  padding: 0 10px;
  border-top: none;
  outline: none;
  color: #fff;
  background-color: #858693;
  box-sizing: border-box;
}

.calc-btns {
  display: grid;
  grid-template-columns: repeat(4, 25%);
}

.calc-btns > button {
  font-size: 14px;
  height: 60px;
  font-family: sans-serif;
  color: #474343;
  background-color: #e0e0e0;
  outline: none;
  border: 0.2px solid #c9c9c9;
}

.calc-btns .elite {
  background-color: #f1913f;
}

.calc-btns #round {
  grid-column: 1/3;
}
