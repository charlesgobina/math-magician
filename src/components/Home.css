@import url('https://fonts.googleapis.com/css2?family=Nunito:wght@300;400;600;800&display=swap');

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Nunito', sans-serif;
}

.homeTitle {
  margin-top: 30px;
  font-size: 30px;
  padding: 10px;
  text-align: left;
  font-family: 'Nunito', sans-serif;
  font-weight: 600;
}

.paraLast {
  font-weight: 600;
  font-size: 22px;
  text-align: left;
}

.homePara {
  display: flex;
  flex-direction: column;
  gap: 40px;
  text-align: left;
  margin: 0 auto;
  padding: 10px;
}

@media screen and (max-width: 400px) {
  .homeTitle,
  .paraLast {
    text-align: center;
  }

  .homeTitle {
    margin-top: 10px;
    font-size: 25px;
  }

  .homePara {
    text-align: center;
  }
}
