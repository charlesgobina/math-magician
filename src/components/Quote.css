@import url('https://fonts.googleapis.com/css2?family=Nunito:wght@300;400;600;800&display=swap');

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Nunito', sans-serif;
}

blockquote {
  font-weight: 600;
  font-size: 18px;
  margin-top: 300px;
  padding: 20px;
  color: #858693;
}

@media screen and (max-width: 400px) {
  blockquote {
    margin-top: 200px;
  }
}
