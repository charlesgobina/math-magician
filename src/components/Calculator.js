/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import './Calculator.css';

export default class Calculator extends Component {
  render() {
    return (
      <div className="wrapper">
        <div className="display">
          <input id="display-main" type="text" />
        </div>
        <div className="calc-btns-wrapper">
          <div className="calc-btns">
            <button className="button" type="button" value="AC" id="clear">AC</button>
            <button className="button" type="button" value="+/-" id="pom">+/-</button>
            <button className="button" type="button" value="%" id="mod">%</button>
            <button className="button elite" type="button" value="/" id="quotient">/</button>
            <button className="button" type="button" value="7" id="seven">7</button>
            <button className="button" type="button" value="8" id="eight">8</button>
            <button className="button" type="button" value="9" id="nine">9</button>
            <button className="button elite" type="button" value="*" id="product">*</button>
            <button className="button" type="button" value="4" id="four">4</button>
            <button className="button" type="button" value="5" id="five">5</button>
            <button className="button" type="button" value="6" id="six">6</button>
            <button className="button elite" type="button" value="-" id="diff">-</button>
            <button className="button" type="button" value="1" id="one">1</button>
            <button className="button" type="button" value="2" id="two">2</button>
            <button className="button" type="button" value="3" id="three">3</button>
            <button className="button elite" type="button" value="+" id="sum">+</button>
            <button className="button" type="button" id="round">0</button>
            <button className="button" type="button" id="dot">.</button>
            <button className="button elite" type="button" id="equals">=</button>
          </div>
        </div>
      </div>
    );
  }
}
