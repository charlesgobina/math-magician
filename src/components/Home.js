import React from 'react';
import './Home.css';

const Home = () => (
  <>
    <h1 className="homeTitle">Welcome to our page!</h1>
    <div className="homePara">
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
        Dolorum consequuntur aliquid pariatur dolorem ut corrupti,
        eserunt eveniet culpa perspiciatis explicabo rem dignissimos
        tempore nemo quae a optio asperiores totam!
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
        Dolorum consequuntur aliquid pariatur dolorem ut corrupti,
        eserunt eveniet culpa perspiciatis explicabo rem dignissimos
        tempore nemo quae a optio asperiores totam!
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
        Dolorum consequuntur aliquid pariatur dolorem ut corrupti,
        eserunt eveniet culpa perspiciatis explicabo rem dignissimos
        tempore nemo quae a optio asperiores totam!
      </p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
        Dolorum consequuntur aliquid pariatur dolorem ut corrupti,
        eserunt eveniet culpa perspiciatis explicabo rem dignissimos
        tempore nemo quae a optio asperiores totam!
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
        Dolorum consequuntur aliquid pariatur dolorem ut corrupti,
        eserunt eveniet culpa perspiciatis explicabo rem dignissimos
        tempore nemo quae a optio asperiores totam!
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
        Dolorum consequuntur aliquid pariatur dolorem ut corrupti,
        eserunt eveniet culpa perspiciatis explicabo rem dignissimos
        tempore nemo quae a optio asperiores totam!
      </p>
      <p className="paraLast">Enjoy!</p>
    </div>
  </>
);

export default Home;
