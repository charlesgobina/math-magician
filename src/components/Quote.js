import React from 'react';
import './Quote.css';

const Quote = () => (
  <>
    <blockquote cite="https://www.childup.com/blog/math-is-not-just-about-numbers-and-equations-its-about-learning-and-understanding#:~:text=As%20William%20Thurston%20said%2C%20%E2%80%9Cmathematics,be%20in%20the%20long%20run.&text=Ignoring%20math%20is%20as%20disabling,not%20knowing%20how%20to%20read.">
      Mathematics is not about numbers, equations, computations,
      or algorithms: It is about understanding. - William Paul Thurston
    </blockquote>
  </>
);

export default Quote;
