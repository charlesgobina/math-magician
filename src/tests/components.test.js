import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
// import Header from '../components/Header';
import Home from '../components/Home';
import Quote from '../components/Quote';

describe('Test UI of the webpage', () => {
  it('Calculator displays on the page', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Quote displays on the page', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Home displays on the page', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
