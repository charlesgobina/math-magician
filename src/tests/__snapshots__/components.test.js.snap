// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UI of the webpage Calculator displays on the page 1`] = `
<div
  className="wrapper"
>
  <div
    className="calcText"
  >
    <p>
      Let us do some math!
    </p>
  </div>
  <div
    className="calcSection"
  >
    <div
      className="display"
    >
      <input
        id="display-main"
        readOnly={true}
        type="text"
        value={0}
      />
    </div>
    <div
      className="calc-btns-wrapper"
    >
      <div
        className="calc-btns"
      >
        <button
          className="button"
          id="clear"
          name="AC"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="button"
          id="pom"
          name="+/-"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="button"
          id="mod"
          name="%"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="button elite"
          id="quotient"
          name="รท"
          onClick={[Function]}
          type="button"
        >
          /
        </button>
        <button
          className="button"
          id="seven"
          name="7"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="button"
          id="eight"
          name="8"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="button"
          id="nine"
          name="9"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="button elite"
          id="product"
          name="x"
          onClick={[Function]}
          type="button"
        >
          *
        </button>
        <button
          className="button"
          id="four"
          name="4"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="button"
          id="five"
          name="5"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="button"
          id="six"
          name="6"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="button elite"
          id="diff"
          name="-"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          className="button"
          id="one"
          name="1"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="button"
          id="two"
          name="2"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="button"
          id="three"
          name="3"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="button elite"
          id="sum"
          name="+"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="button"
          id="round"
          name="0"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="button"
          id="dot"
          name="."
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="button elite"
          id="equals"
          name="="
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test UI of the webpage Home displays on the page 1`] = `
Array [
  <h1
    className="homeTitle"
  >
    Welcome to our page!
  </h1>,
  <div
    className="homePara"
  >
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolorum consequuntur aliquid pariatur dolorem ut corrupti, eserunt eveniet culpa perspiciatis explicabo rem dignissimos tempore nemo quae a optio asperiores totam! Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolorum consequuntur aliquid pariatur dolorem ut corrupti, eserunt eveniet culpa perspiciatis explicabo rem dignissimos tempore nemo quae a optio asperiores totam! Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolorum consequuntur aliquid pariatur dolorem ut corrupti, eserunt eveniet culpa perspiciatis explicabo rem dignissimos tempore nemo quae a optio asperiores totam!
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolorum consequuntur aliquid pariatur dolorem ut corrupti, eserunt eveniet culpa perspiciatis explicabo rem dignissimos tempore nemo quae a optio asperiores totam! Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolorum consequuntur aliquid pariatur dolorem ut corrupti, eserunt eveniet culpa perspiciatis explicabo rem dignissimos tempore nemo quae a optio asperiores totam! Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolorum consequuntur aliquid pariatur dolorem ut corrupti, eserunt eveniet culpa perspiciatis explicabo rem dignissimos tempore nemo quae a optio asperiores totam!
    </p>
    <p
      className="paraLast"
    >
      Enjoy!
    </p>
  </div>,
]
`;

exports[`Test UI of the webpage Quote displays on the page 1`] = `
<blockquote
  cite="https://www.childup.com/blog/math-is-not-just-about-numbers-and-equations-its-about-learning-and-understanding#:~:text=As%20William%20Thurston%20said%2C%20%E2%80%9Cmathematics,be%20in%20the%20long%20run.&text=Ignoring%20math%20is%20as%20disabling,not%20knowing%20how%20to%20read."
>
  Mathematics is not about numbers, equations, computations, or algorithms: It is about understanding. - William Paul Thurston
</blockquote>
`;
